lib/hsa/mcwamp_hsa.cpp

```diff
@@ -3171,6 +3191,21 @@ HSADispatch::setLaunchConfiguration(int dims, size_t *globalDims, size_t *localD
       workgroup_total_size = workgroup_size[0] * workgroup_size[1] * workgroup_size[2];
     }
 
+    static const size_t num_work_items_per_simd = 64;
+    static const size_t num_simds_per_cu = 4;
+    static const size_t num_work_items_per_cu = num_work_items_per_simd * num_simds_per_cu;
+    size_t num = (workgroup_total_size + num_work_items_per_cu - 1) / num_work_items_per_cu;
+    size_t num_sgprs = num * kernel->wavefront_sgpr_count;
+    size_t num_vgprs = num * kernel->workitem_vgpr_count;
+    // Every wavefront is allocated up to a maximum of 104 SGPRs on the GPUs prior to VI and 102 SGPRs on VI.
+    if (num_sgprs > 102) {
```

Comment:
@weixingzhang  could you revise this hard-coded value as a macro? Also it would be better if we use ISA version (ex: GFX8), instead of ASIC family names (ex: VI).